lib/elixir/src/elixir_erl.erl

```diff
@@ -461,6 +466,12 @@ take_type_spec(Data, Key) ->
     [] -> []
   end.
 
+validate_spec_for_existing_function(#{definitions := Defs} = Map, NameAndArity, Line) ->
+  case lists:keymember(NameAndArity, 1, Defs) of
+    false -> form_error(Map#{line := Line}, {spec_for_undefined_function, NameAndArity});
```

Comment:
I would prefer to pass `#{line := line, file := ?m(Map, file)}` as it is more explicit since form_error was not really designed to handle the `ModuleMap`.